src/Evolution/DgSubcell/Actions/Initialize.hpp

```diff
@@ -198,17 +204,33 @@ struct Initialize {
         },
         make_not_null(&box));
     if (cell_is_troubled) {
-      // Set variables on subcells.
-      if constexpr (System::has_primitive_and_conservative_vars) {
-        db::mutate<typename System::primitive_variables_tag>(
-            make_not_null(&box), [&subcell_mesh](const auto prim_vars_ptr) {
-              prim_vars_ptr->initialize(subcell_mesh.number_of_grid_points());
-            });
+      if constexpr (UseNumericInitialData) {
```

Comment:
Can you more easily check if the `AnalyticData` or `AnalyticSolution` tag is in the box? Then we don't need to add a template parameter to this action. I honestly don't love how invasive the numerical initial data is currently (not just in this PR). It causes an avalanche of template parameters to propagate through a lot of the code. It's almost as invasive as DG-FD, I find.

Reply:
done